server/auth/authService.js

```diff
@@ -0,0 +1,56 @@
+const jwt = require('jsonwebtoken');
+const User = require('../db/db').User;
+const config = require('../.config.js');
+const expressJwt = require('express-jwt');
+
+const EXPIRY = 60 * 60 * 5;
+const SECRET = config.SESSION_SECRET;
+const validateJwt = expressJwt({ secret: SECRET });
+
+// Include access_token query param in req.header for validateJwt
+const accessTokenHeader = (req, res, next) => {
+  if (req.query && req.query.hasOwnPropery('access_token')) {
+    req.headers.authorization = `Bearer ${req.query.access_token}`;
+  }
+  next();
+};
+
+// Convert req.user after validateJwt
+// Object with an id to User instance from database
+const populateReqUser = (req, res, next) => {
+  User.find({ where: { id: req.user.id } })
+    .then(user => {
+      if (!user) {
+        return res.status(401).end();
+      }
+      req.user = user;
+      return next();
+    })
+    .catch(err => next(err));
+};
+
+// TODO: Confirm this combination works
+//       Should validate tokens, attach user to a request
+const isAuthenticated = (req, res, next) => {
+  accessTokenHeader(req, res, next)
+    .validateJwt()
+    .populateReqUser(req, res, next);
+};
+
+const signToken = id => jwt.sign({ id: id }, SECRET, { expiresIn: EXPIRY });
+
+const setTokenCookie = (req, res) => {
+  if (!req.user) {
+    return res.status(404).send('User not logged in, please try again.');
+  }
+  res.cookie('token', signToken(req.user.id));
+  return res.redirect('/');
```

Comment:
Failed authentication will be handled on the client side via a pop-up or client side route.